server/routes/slack/slack_events.py

```
@@ -36,9 +36,14 @@
 def slack_events():
     """
     Slack Events API webhook endpoint.
-    Handles @Aurora mentions and routes them to the chat system.
+    Handles incoming messages with @Aurora mentions.
     """
     try:
+        # Reject Slack retries - we already processed the original
+        if request.headers.get('X-Slack-Retry-Num'):
+            logger.info(f"Ignoring Slack retry #{request.headers.get('X-Slack-Retry-Num')}")
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)